openmls/src/treesync/node/leaf_node.rs

```diff
@@ -263,53 +285,122 @@ impl LeafNode {
+        encryption_keys: &[EncryptionKey],
+        members_supported_credentials: &[&[CredentialType]],
+        currently_in_use: &[CredentialType],
+    ) -> Result<&Self, LeafNodeValidationError> {
+        self.validate(
+            version,
+            ciphersuite,
+            required_capabilities,
+            signature_keys,
+            encryption_keys,
+            members_supported_credentials,
+            currently_in_use,
+        )?;
 
+        // If the LeafNode appears in an Update proposal, verify that leaf_node_source is set to update.
         match self.payload.leaf_node_source {
             LeafNodeSource::Update => Ok(self),
             _ => Err(LeafNodeValidationError::InvalidLeafNodeSource),
         }
     }
 
     /// Validate the leaf node in the context of a commit.
-    // TODO(#1186)
-    #[allow(unused)]
-    pub(crate) fn validate_in_commit(&self) -> Result<&Self, LeafNodeValidationError> {
-        // TODO(#1186)
-        // self.validate()?;
+    pub(crate) fn validate_in_commit<'a>(
+        &self,
+        version: ProtocolVersion,
+        ciphersuite: Ciphersuite,
+        required_capabilities: impl Into<Option<&'a RequiredCapabilitiesExtension>>,
+        signature_keys: &[SignaturePublicKey],
+        encryption_keys: &[EncryptionKey],
+        members_supported_credentials: &[&[CredentialType]],
+        currently_in_use: &[CredentialType],
+    ) -> Result<&Self, LeafNodeValidationError> {
+        self.validate(
+            version,
+            ciphersuite,
+            required_capabilities,
+            signature_keys,
+            encryption_keys,
+            members_supported_credentials,
+            currently_in_use,
+        )?;
 
+        // If the LeafNode appears in the leaf_node value of the UpdatePath in a Commit, verify
+        // that leaf_node_source is set to commit.
         match self.payload.leaf_node_source {
             LeafNodeSource::Commit(_) => Ok(self),
             _ => Err(LeafNodeValidationError::InvalidLeafNodeSource),
         }
     }
 
     /// Basic validation of leaf node called in all `validate_in_*` methods.
-    // TODO(#1186)
-    #[allow(unused)]
-    fn validate<'a>(
+    pub(crate) fn validate<'a>(
         &self,
+        version: ProtocolVersion,
+        ciphersuite: Ciphersuite,
         required_capabilities: impl Into<Option<&'a RequiredCapabilitiesExtension>>,
         signature_keys: &[SignaturePublicKey],
         encryption_keys: &[EncryptionKey],
         members_supported_credentials: &[&[CredentialType]],
         currently_in_use: &[CredentialType],
     ) -> Result<&Self, LeafNodeValidationError> {
-        self.validate_required_capabilities(required_capabilities)?
-            .validate_that_capabilities_contain_extension_types()?
+        // The client verifies the validity of a LeafNode using the following steps:
+        self
+            // Verify that the credential in the LeafNode is valid as described in Section 5.3.1.
```

Comment:
The AS verification has to happen on the outside. Maybe add a comment to the entry points to the validation that all the AS validation must have happened before.